catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/InputTransformerIds.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InputTransformerIds {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(InputTransformerIds.class);
+
+  private static Path transformerFolder = Paths.get("etc", "transformers");
+
+  private InputTransformerIds() {}
+
+  public static Set<String> getTransformerIds() {
+    File[] transformerFiles = transformerFolder.toFile().listFiles();
+    if (transformerFiles == null || transformerFiles.length == 0) {
+      LOGGER.debug("No Input Transformer json files found in {}", transformerFolder);
+      return Collections.emptySet();
+    }
+
+    Set<String> allTransformerIds = new HashSet<>();
+    Gson gson = new Gson();
+    for (File transformerFile : transformerFiles) {
+      try {
+        String json = IOUtils.toString(transformerFile.toURI(), Charset.defaultCharset());
+        List<String> filesTransformerIds =
+            gson.fromJson(json, TransformerJsonFile.class).getTransformerIds();
```

Comment:
❓ What is the point for the `TransformerJsonFile` if at the end of the day, you are just deserializing a list of strings?
Simply do:
`gson.fromJson(json, new TypeToken<List<String>>() {}.getType())`